jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java

```diff
@@ -0,0 +1,168 @@
+        this.blacklistedPackagesList = blacklistedPackagesList;
+    }
+
+    /**
+     * Get All monitors and store metric metricClass to compositeData[].
+     *
+     * @throws OpenDataException creating compositetype fail
+     */
+    private void scan() {
+        final Set<Class<?>> monitorClasses = new HashSet<>();
+        final ClassGraph classGraph = new ClassGraph();
+        classGraph.enableAnnotationInfo().ignoreClassVisibility().blacklistLibOrExtJars()
+                .removeTemporaryFilesAfterScan()
+                .overrideClassLoaders(this.getClass().getClassLoader()).ignoreParentClassLoaders();
+        if (whitelistedPackagesList.size() > 0) {
+            classGraph.whitelistPackages(whitelistedPackagesList.toArray(new String[0]));
+        }
+        if (blacklistedPackagesList.size() > 0) {
+            classGraph.blacklistPackages(blacklistedPackagesList.toArray(new String[0]));
+        }
+        final ScanResult scanResult = classGraph.scan();
+        final ClassInfoList classInfoList = scanResult.getClassesWithAnnotation(MetricClass.class.getName());
+        monitorClasses.addAll(classInfoList.loadClasses(true));
+
+        final List<CompositeData> metricClassCompositesList = new ArrayList<>();
+        for (Class<?> monitorClass : monitorClasses) {
+            try {
+                final CompositeData metricClassComposite = new MonitorCompositeDataBuilder(monitorClass).getMetricClassData();
+                metricClassCompositesList.add(metricClassComposite);
+            } catch (OpenDataException ox) {
+                logger.error("Getting monitor metricaAnnotation failed: " + monitorClass.getName());
+            }
+        }
+
+        final int monitorCount = metricClassCompositesList.size();
+        final String[] monitorKey = new String[monitorCount];
+        final String[] monitorKeyDescription = new String[monitorCount];
+        @SuppressWarnings("rawtypes")
+        final OpenType[] monitorType = new OpenType[monitorCount];
+        final CompositeData[] metricClassComposites = metricClassCompositesList.toArray(new CompositeData[0]);
+        for (int j = 0; j < metricClassComposites.length; j++) {
+            final CompositeData metricClassComposite = metricClassComposites[j];
+            monitorKey[j] = metricClassComposite.getCompositeType().getTypeName();
+            monitorKeyDescription[j] = metricClassComposite.getCompositeType().getDescription();
+            monitorType[j] = metricClassComposite.getCompositeType();
+        }
+
+        try {
+            final CompositeType allMonitorCompositeType = new CompositeType("Monitor Metric", "Monitor Metric Info", monitorKey,
+                    monitorKeyDescription, monitorType);
+            allMonitorCompositeData = new CompositeDataSupport(allMonitorCompositeType, monitorKey, metricClassComposites);
+        } catch (final OpenDataException ox) {
+            logger.error("Creating CompositeData failed", ox);
```

Comment:
What happens on jmx if we fail here?

Reply:
Then scan failed and wait for next scan in 1 min